R/pkg/R/DataFrame.R

```diff
@@ -392,7 +392,11 @@ setMethod("coltypes",
                 }
 
                 if (is.null(type)) {
-                  stop(paste("Unsupported data type: ", x))
+                  specialtype <- specialtypeshandle(x)
```

Comment:
here it looks like we are expecting `type` to be R native type, but seems like `specialtypeshandle` is returning `double` which is not a R native type?